### PR TITLE
Adding AWS SQS producer

### PR DESCRIPTION
**Pull request by vikrant2mahajan**
In some cases where streaming or ordering is not important a basic sqs can be used. A cheaper and lighter alternative.

Update : With SQS fifo support coming in , it can also support ordering now in case needed.

### CODE REVIEW

**Comment by vikrant2mahajan** [comment]
@osheroff  what is the process of getting a PR reviewed and merged for collaboration

**Comment by osheroff** [comment]
Pretty loose, just a plus one from a maintainer.  I'll take a look when I get a sec, or maybe Tim can.

**Comment by vikrant2mahajan** [comment]
@timbertson can you have a look the pull request

File: src/main/java/com/zendesk/maxwell/MaxwellContext.java [hunk 1]
```diff
@@ -331,7 +331,10 @@ public AbstractProducer getProducer() throws IOException {
 			case "kinesis":
 				this.producer = new MaxwellKinesisProducer(this, this.config.kinesisStream);
 				break;
-			case "profiler":
+			case "sqs":
+				this.producer = new MaxwellSQSProducer(this, this.config.sqsQueueUri);
+				break;
+				case "profiler":
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellContext.java]: whitespace issue here

**Comment by osheroff** [comment]
needs a bit of documentation in the "producers" section of the docs

File: src/main/java/com/zendesk/maxwell/MaxwellContext.java (hunk 1 above)
Comment (vikrant2mahajan) [inline src/main/java/com/zendesk/maxwell/MaxwellContext.java]: fixed the whitespace issue

**Comment by vikrant2mahajan** [comment]
@osheroff added the documentation as well in producers section

**Comment by vikrant2mahajan** [comment]
@osheroff any update
--

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 2]
```diff
@@ -153,6 +155,7 @@ protected OptionParser buildOptionParser() {
 		parser.accepts( "kafka_version", "use kafka 0.8, 0.9, 0.10, 0.10.1, or 0.10.2 producer (default 0.9)");
 
 		parser.accepts( "kinesis_stream", "kinesis stream name").withOptionalArg();
+		parser.accepts("sqs_queue_uri", "SQS Queue uri").withOptionalArg();
```
Comment (liulikun) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: The value for `sqs_queue_uri` needs to be supplied? In that case, you'll need to use `withRequiredArg` here.

File: src/main/java/com/zendesk/maxwell/producer/MaxwellSQSProducer.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		}
+	};
+
+	@Override
+	public void onSuccess(SendMessageRequest request, SendMessageResult result) {
+		if (logger.isDebugEnabled()) {
+			logger.debug("-> Message id:" + result.getMessageId() + ", sequence number:" + result.getSequenceNumber());
+			logger.debug("   " + json);
+			logger.debug("   " + position);
+			logger.debug("");
```
Comment (liulikun) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellSQSProducer.java]: Can we log the debug message in one line? Those multiline messages could be interleaved with multiple processes.

File: docs/docs/producers.md [hunk 4]
```diff
@@ -117,3 +117,17 @@ Set the output stream in `config.properties` by setting the `kinesis_stream` pro
...
+
+See the [AWS docs](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html) on how to setup the IAM user with the Default Credential Provider Chain.
+
+In case you need to set up a different region also along with credentials then default one, see the [AWS docs](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-credentials.html#setup-credentials-setting-region).
+
+### SQS Options
+***
+Set the output queue in the `config.properties` by setting the `sqs_queue_uri` property to full SQS queue uri from aws console.
+
+The producer uses the [AWS SQS SDK](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/sqs/AmazonSQSClient.html).
```
Comment (liulikun) [inline docs/docs/producers.md]: Nitpicks on the doc: `sqs` -> `SQS`, `aws` -> `AWS` (avoid mixed casing for product names)

**Comment by liulikun** [comment]
@vikrant2mahajan sorry for the delayed review. Can you resolve the merge conflicts? Let me know if you need any help. I'm hoping to merge this soon.

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java (hunk 2 above)
Comment (vikrant2mahajan) [reply src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: resolved

File: src/main/java/com/zendesk/maxwell/producer/MaxwellSQSProducer.java (hunk 3 above)
Comment (vikrant2mahajan) [reply src/main/java/com/zendesk/maxwell/producer/MaxwellSQSProducer.java]: resolved

File: docs/docs/producers.md (hunk 4 above)
Comment (vikrant2mahajan) [reply docs/docs/producers.md]: resolved

**Comment by vikrant2mahajan** [comment]
@liulikun  @osheroff  Done the code review changes as requested can we get this merged now ? Let me know any changes needed

**Comment by liulikun** [comment]
Thanks for contributing. Looks good! 👍

**Pull request merged**